samples/dynamic-headers/custom-visual/src/visual.ts

```diff
@@ -0,0 +1,127 @@
+import Tabulator from 'tabulator-tables';
+import { VisualSettings } from "./settings";
+
+export class Visual implements IVisual {
+    private target: HTMLElement;
+    private settings: VisualSettings;
+    private dynamic_table_html: HTMLElement;
+    private error_message_html: HTMLElement;
+
+    constructor(options: VisualConstructorOptions) {  
+        this.target = options.element;
+        
+        if (document) {
+            this.dynamic_table_html = document.createElement("div");
+            this.dynamic_table_html.setAttribute("id", "dynamic-header-table");
+
+            this.error_message_html = document.createElement("p"); 
+            this.error_message_html.style.display = "none";
+
+            this.target.appendChild(this.dynamic_table_html);
+            this.target.appendChild(this.error_message_html);
+        }
+    }
+ 
+    public update(options: VisualUpdateOptions) {
+        var tableColumns = options.dataViews[0].table.columns;        
+        var tableRows = options.dataViews[0].table.rows;
+
+        // Rewrite columns index based on their order
+        tableColumns.forEach( (tableColumn, index) =>  tableColumn.index = index );
+        
+        // Get all different data roles
+        var staticTitleColumns = tableColumns
+            .filter(staticTitleColumn => staticTitleColumn.roles.static_title_columns == true)
+            .map(staticTitleColumn => {
+                return { 
+                    index: staticTitleColumn.index,
+                    roleIndex: staticTitleColumn['rolesIndex'].static_title_columns[0], 
+                    displayName: staticTitleColumn.displayName
+                }
+            })
+            .sort((a, b) => a.roleIndex - b.roleIndex);
+        
+        var dynamicTitleColumns = tableColumns
+            .filter(dynamicTitleColumn => dynamicTitleColumn.roles.dynamic_title_columns == true)
+            .map(dynamicTitleColumn => {  
+                return {
+                    index: dynamicTitleColumn.index,
+                    roleIndex: dynamicTitleColumn['rolesIndex'].dynamic_title_columns[0], 
+                    displayName: dynamicTitleColumn.displayName
+                } 
+            })
+            .sort((a, b) => a.roleIndex - b.roleIndex);
+
+        var dynamicHeaderTitles = tableColumns
+            .filter(dynamicHeaderTitle => dynamicHeaderTitle.roles.dynamic_header_titles == true)
+            .map(dynamicHeaderTitle => { 
+                return { 
+                    index: dynamicHeaderTitle.index,
+                    roleIndex: dynamicHeaderTitle['rolesIndex'].dynamic_header_titles[0],
+                } 
+            })
+            .sort((a, b) => a.roleIndex - b.roleIndex);
+
+        // Validate common issues
+        // Different number of fields dragged into the header and values roles
+        if(dynamicTitleColumns.length != dynamicHeaderTitles.length){
```

Comment:
NIT: Add space after `if` and after `(`

Should look like this:

```
if (dynamicTitleColumns.length != dynamicHeaderTitles.length) {
```

Please check if there are other places with similar issues.